### PR TITLE
Add Support for v2.9.1 MSP PR 866 in BF

### PR DESCRIPTION
**Pull request by borisbstyle**
Add support for v2.9.1 maintanance release.

@mikeller 
What do you think?

### CODE REVIEW

File: js/msp.js [hunk 1]
```diff
@@ -297,10 +297,12 @@ var MSP = {
...
                 CONFIG.profile = data.getUint8(10);
                 CONFIG.cpuload = data.getUint16(11, 1);
-                CONFIG.numProfiles = data.getUint8(13);
-                CONFIG.rateProfile = data.getUint8(14);
+                if (semver.gt(CONFIG.flightControllerVersion, "2.9.1")) {
+                    CONFIG.numProfiles = data.getUint8(13);
+                    CONFIG.rateProfile = data.getUint8(14);
 
-                TABS.pid_tuning.checkUpdateProfile(true);
+                    TABS.pid_tuning.checkUpdateProfile(true);
```
Comment (mikeller) [inline js/msp.js]: This shouldn't be conditional. Profile switching is supported by earlier versions as well, and there are guards for the rateprofile related functionality inside the method.

File: js/msp.js [hunk 2]
```diff
@@ -407,9 +409,9 @@ var MSP = {
                 }
                 if (semver.gte(CONFIG.apiVersion, "1.10.0")) {
                     RC_tuning.RC_YAW_EXPO = parseFloat((data.getUint8(offset++) / 100).toFixed(2));
-                    if (semver.gte(CONFIG.apiVersion, "1.20.0")) {
+                    if (semver.gte(CONFIG.flightControllerVersion, "2.9.1")) {
```
Comment (mikeller) [inline js/msp.js]: We should either go for all checks apiVersion or all checks flightControllerVersion in one file. Having a complete mess of both makes this a nightmare to debug.

File: js/msp.js [hunk 3]
```diff
@@ -1001,15 +1003,16 @@ var MSP = {
                 }
                 break;
             case MSP_codes.MSP_SPECIAL_PARAMETERS:
```
Comment (mikeller) [inline js/msp.js]: I thought with 2.9.1 we'd be able to get rid of `MSP_SPECIAL_PARAMETERS` altogether?

File: js/msp.js [hunk 4]
```diff
@@ -1001,15 +1003,16 @@ var MSP = {
...
-                      RX_CONFIG.airModeActivateThreshold = data.getUint16(offset, 1);
-                      offset += 2;
-                      RX_CONFIG.rcSmoothInterval = data.getUint8(offset++, 1)
-                      SPECIAL_PARAMETERS.escDesyncProtection = data.getUint16(offset, 1);
-                 }
-                 break;
+                var offset = 0;
+                if (semver.gte(CONFIG.flightControllerVersion, "2.8.0")) {
+                    RC_tuning.rcYawRate = parseFloat((data.getUint8(offset++) / 100).toFixed(2));
+                } else if (semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
```
Comment (mikeller) [inline js/msp.js]: We shouldn't be checking for 2.8.2, as this was never released. Also, please use either apiVersion or flightControllerVersion.

File: js/msp.js [hunk 5]
```diff
@@ -1695,11 +1698,13 @@ MSP.crunch = function (code) {
             }
             break;
         case MSP_codes.MSP_SET_SPECIAL_PARAMETERS:
```
Comment (mikeller) [inline js/msp.js]: Doesn't your PR for 2.9.1 remove this command?

File: js/msp.js (hunk 1 above)
Comment (borisbstyle) [inline js/msp.js]: Its not in the MSP of the 2.9.1 for MSP_STATUS_EX, but could be added.

Like I say this is a maintanance release not fully functional like 3.0. 2.9.0 didnt have this either

File: js/msp.js (hunk 2 above)
Comment (borisbstyle) [inline js/msp.js]: Lets do later cleanup for this. For 2.9.1 this is needed as it uses same API as 2.9.0.

We need to get rid of API stuff I think

Comment (mikeller) [inline js/msp.js]: Ok, then please remove it everywhere. Just creating more mess is not helping.

File: js/msp.js (hunk 3 above)
Comment (borisbstyle) [inline js/msp.js]: its gone in the latest commit. Missed that.

File: js/msp.js (hunk 1 above)
Comment (mikeller) [inline js/msp.js]: `CONFIG.profile`, and we need to act on that one. Otherwise a user might be editing a profile, and at the same time switch profiles on the FC (through the TX), and then save the edits to the changed profile.

File: js/msp.js (hunk 2 above)
Comment (borisbstyle) [inline js/msp.js]: @mikeller 
Its not part of this PR.

Comment (borisbstyle) [inline js/msp.js]: Needs seperate cleanup PR, which I may do

File: js/msp.js (hunk 1 above)
Comment (mikeller) [inline js/msp.js]: Will fix the broken profile switching for 2.9.1 tonight.

File: js/msp.js (hunk 3 above)
Comment (mikeller) [inline js/msp.js]: It seems to be still in there, looks like you missed that commit.
